File: two-sum/yd.py
class Solution(object):
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        nmap = {}
        for i in range(len(nums)):
            rest = target - nums[i]
            if rest in nmap:
                return [nmap[rest], i]
            nmap[nums[i]] = i
        return []

File: two-sum/yd.java
public class Solution {
    public int[] twoSum(int[] numbers, int target) {
        if (numbers == null || numbers.length < 2) {
            return null;
        }
        Map<Integer, Integer> map = new HashMap<Integer, Integer>();
        for (int i = 0; i < numbers.length; i++) {
            int rest = target - numbers[i];
            if (map.containsKey(rest)) {
                return new int[]{map.get(rest), i};
            }
            map.put(numbers[i], i);
        }
        return null;
    }
}
